### DISCUSSION TITLE
yolov5 does not work for webcam?

### DISCUSSION THREAD

**Discussion by StanleyData** [ultralytics/yolov5#10306] [Q&A]
Hey everyone! I am running detect.py in the terminal using: python detect.py --source 0. I am getting an annoying error message

  File "D:\Pycharm stuff\PycharmProjects\tuxiang2\yolov5-5.0\detect.py", line 182, in <module>
    detect()
  File "D:\Pycharm stuff\PycharmProjects\tuxiang2\yolov5-5.0\detect.py", line 55, in detect
    dataset = LoadStreams(source, img_size=imgsz, stride=stride)
  File "D:\Pycharm stuff\PycharmProjects\tuxiang2\yolov5-5.0\utils\datasets.py", line 284, in __init__
    if 'youtube.com/' in url or 'youtu.be/' in url:  # if source is YouTube video
TypeError: argument of type 'int' is not iterable

can anyone help me? thanks so much

**Comment by glenn-jocher** [comment]
@StanleyData your code is out of date. To update:

- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>
